dedupe/convenience.py

```diff
@@ -88,101 +88,101 @@ def randomPairsWithReplacement(n_records: int, sample_size: int) -> IndicesItera
+        n_distinct = len(deduper.training_pairs['distinct']) + sum(
+            label == 'distinct' for _, label in labeled
+        )
 
-        for pair in record_pair:
+        for record in record_pair:
             for field in fields:
-                line = "%s : %s" % (field, pair[field])
-                print(line, file=sys.stderr)
-            print(file=sys.stderr)
-
-        print("{0}/10 positive, {1}/10 negative".format(n_match, n_distinct),
-              file=sys.stderr)
-        print('Do these records refer to the same thing?', file=sys.stderr)
+                line = '%s : %s' % (field, record[field])
+                _print(line)
+            _print()
+        _print('{0}/10 positive, {1}/10 negative'.format(n_match, n_distinct))
+        _print('Do these records refer to the same thing?')
 
         valid_response = False
         user_input = ''
         while not valid_response:
-            if examples_buffer:
-                prompt = '(y)es / (n)o / (u)nsure / (f)inished / (p)revious'
+            if labeled:
+                _print('(y)es / (n)o / (u)nsure / (f)inished / (p)revious')
                 valid_responses = {'y', 'n', 'u', 'f', 'p'}
             else:
-                prompt = '(y)es / (n)o / (u)nsure / (f)inished'
+                _print('(y)es / (n)o / (u)nsure / (f)inished')
                 valid_responses = {'y', 'n', 'u', 'f'}
-
-            print(prompt, file=sys.stderr)
             user_input = input()
             if user_input in valid_responses:
                 valid_response = True
 
         if user_input == 'y':
-            examples_buffer.insert(0, (record_pair, 'match'))
+            labeled.insert(0, (record_pair, 'match'))
         elif user_input == 'n':
-            examples_buffer.insert(0, (record_pair, 'distinct'))
+            labeled.insert(0, (record_pair, 'distinct'))
         elif user_input == 'u':
-            examples_buffer.insert(0, (record_pair, 'uncertain'))
+            # See https://github.com/dedupeio/dedupe/issues/984 for reasoning
+            labeled.insert(0, (record_pair, 'match'))
+            labeled.insert(0, (record_pair, 'distinct'))
         elif user_input == 'f':
-            print('Finished labeling', file=sys.stderr)
+            _print('Finished labeling')
             finished = True
         elif user_input == 'p':
             use_previous = True
-            uncertain_pairs.append(record_pair)
-
-        if len(examples_buffer) > buffer_len:
-            record_pair, label = examples_buffer.pop()
-            if label in {'distinct', 'match'}:
-
-                examples: TrainingData
-                examples = {'distinct': [],
-                            'match': []}
-                examples[label].append(record_pair)
-                deduper.mark_pairs(examples)
+            unlabeled.append(record_pair)
 
-    for record_pair, label in examples_buffer:
-        if label in ['distinct', 'match']:
+        while len(labeled) > buffer_len:
```

Comment:
we'll also need to do this a bit differently, since unsure adds to elements to labeled